src/system.rs

```diff
@@ -13,6 +14,174 @@ use std::convert::TryInto;
 use std::io::ErrorKind;
 use std::rc::Rc;
 
+pub type SharedContext = Rc<RefCell<Context>>;
+
+/// Defines a [Context] struct which stores information on parent types
+/// while loading a [Schema] with types which is used to resolve self referencing schema types
+#[derive(Debug, Clone)]
+pub struct Context {
```

Comment:
`Context` is a reasonable name, but I think we can be more specific: `PendingTypes` or `ParentTypes` give a clearer sense of its role, IMO.

Reply:
Sure. I think `PendingTypes` sounds like a good name for this.